presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -73,6 +76,57 @@ public static GeometryTypeName valueOf(String type)
         }
     }
 
+    // Copy of com.esri.core.geometry.Interop.translateFromAVNaN
+    private static double translateFromAVNaN(double n)
+    {
+        return n < -1.0E38D ? (0.0D / 0.0) : n;
+    }
+
+    @Nullable
+    public static Envelope deserializeEnvelope(Slice shape)
+    {
+        if (shape == null) {
+            return null;
+        }
+        BasicSliceInput input = shape.getInput();
+
+        input.readInt(); // spatial reference ID
+
+        Envelope overallEnvelope = null;
+        while (input.available() > 0) {
+            int length = input.readInt();
+            ByteBuffer buffer = input.readSlice(length).toByteBuffer().slice().order(LITTLE_ENDIAN);
+            int type = buffer.getInt();
+            Envelope envelope = null;
+            if (type == 1) {    // point
+                double x = buffer.getDouble();
+                double y = buffer.getDouble();
+                if (!Double.isNaN(x) && !Double.isNaN(translateFromAVNaN(x))) {
+                    envelope = new Envelope(x, y, x, y);
+                }
+            }
+            else {
+                double xmin = buffer.getDouble();
+                double ymin = buffer.getDouble();
+                double xmax = buffer.getDouble();
+                double ymax = buffer.getDouble();
+                if (!Double.isNaN(xmin) && !Double.isNaN(translateFromAVNaN(xmin))) {
+                    envelope = new Envelope(xmin, ymin, xmax, ymax);
+                }
+            }
+            if (envelope != null) {
```

Comment:
I understand `envelope` might be `null` if some coordinate is `NaN`. In what case can this happen? Don't we loose some valuable info this way?

Reply:
NaN values are used when serializing empty geometries, e.g. 'POINT EMPTY', 'POLYGON EMPTY'.